internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -279,26 +295,46 @@ ExecExprVisitor::ExecBinaryArithOpEvalRangeVisitorDispatcher(BinaryArithOpEvalRa
         case OpType::Equal: {
             switch (arith_op) {
                 case ArithOpType::Add: {
+                    auto index_func = [val, right_operand](Index* index, size_t offset) {
+                        auto x = index->Reverse_Lookup(offset);
+                        return (x + right_operand) == val;
+                    };
                     auto elem_func = [val, right_operand](T x) { return ((x + right_operand) == val); };
-                    return ExecDataRangeVisitorImpl<T>(expr.field_id_, elem_func);
+                    return ExecDataRangeVisitorImpl<T>(expr.field_id_, index_func, elem_func);
```

Comment:
I don't think it's a good way that all arithmetic-related expressions should executed by the index. Maybe we can check if raw data of involved fields were loaded later. If loaded, I think it's better to use bruteforce-scan here. Otherwise then should we check if the index can be used to do retrieve?

Reply:
We can optimize this later.

Reply:
> I don't think it's a good way that all arithmetic-related expressions should executed by the index. Maybe we can check if raw data of involved fields were loaded later. If loaded, I think it's better to use bruteforce-scan here. Otherwise then should we check if the index can be used to do retrieve?

This is same to the implementation now